saleor/webhook/payloads.py

```diff
@@ -839,6 +835,9 @@ def generate_product_variant_payload(
     requestor: Optional["RequestorOrLazyObject"] = None,
     with_meta: bool = True,
 ):
+    product_variants = ProductVariant.objects.prefetched_for_webhook().filter(
```

Comment:
Keep in mind, that this function is called inside `generate_product_payload`. It means that you will make a prefetch one by one for each variant assigned to the product. It won't give us anything. 
Instead of calling this always, we should have a helper method that will confirm if the prefetched is applied or not. This should solve the problem of calling this function at different stage of creating payload.

Something like this:
```python
# you can apply different name, I just pick up the first one
def should_apply_prefetch(object, requested_prefetches: list[str]):
    if prefetches:=getattr(product_variant, "_prefetched_objects_cache",None):
         # requested_prefetches should exists in prefeteches to return False
    return True
```
`_prefetched_objects_cache` will give you something dict in below form:
```
{'attributes': <QuerySet []>,
 'variant_media': <QuerySet [<VariantMedia: VariantMedia object (58)>]>,
 'stocks': <StockQuerySet [<Stock: Stock object (457)>, <Stock: Stock object (458)>, <Stock: Stock object (459)>, <Stock: Stock object (460)>, <Stock: Stock object (461)>, <Stock: Stock object (462)>, <Stock: Stock object (463)>, <Stock: Stock object (464)>]>,
 'channel_listings': <ProductVariantChannelListingQuerySet [<ProductVariantChannelListing: ProductVariantChannelListing object (342)>, <ProductVariantChannelListing: ProductVariantChannelListing object (343)>]>}
 ```

Reply:
Ahh, my bad. We can ignore it. I thought that the function accepts a single variant.